### ISSUE TITLE
fhir:link missing from MedicationStatement.subject

### ISSUE THREAD

**Issue by dbooth-boston** [w3c/hcls-fhir-rdf#121]
**Turtle filename**
This R4 example has a fhir:link inside the fhir:MedicationStatement.subject:
https://hl7.org/fhir/R4B/medicationstatementexample1.ttl.html

But it is missing in the R5 example.  Look for fhir:subject in this example:
https://build.fhir.org/medicationstatementexample1.ttl.html

**FHIR Resource**
https://build.fhir.org/medicationstatement.html#af

**ShEx shape**
https://build.fhir.org/medicationstatement.shex.html

**Description**
The turtle probably should have the fhir:link or something equivalent, but we should discuss.

This issue is related to https://github.com/w3c/hcls-fhir-rdf/issues/120

**Comment by dbooth-boston** [comment]
I just checked, and this is not fixed by https://github.com/HL7/kindling/pull/148

**Comment by dbooth-boston** [comment]
@balhoff

**Comment by balhoff** [comment]
@dbooth-boston can you assign to me?

**Comment by dbooth-boston** [comment]
@tmprd Do you think this is a change needed in Kindling?

**Comment by tmprd** [comment]
No, I think this needs to be changed in https://github.com/hapifhir/org.hl7.fhir.core

This might be the method used to add`fhir:link`: https://github.com/hapifhir/org.hl7.fhir.core/blob/0ebf628becc39dfeba744a25e82e68b6508e5674/org.hl7.fhir.r5/src/main/java/org/hl7/fhir/r5/elementmodel/TurtleParser.java#L372C18-L372C35

**Comment by balhoff** [comment]
I think this may be the same issue causing the root node in the R5 examples to be a blank node, when it had an IRI in R4.

E.g. `<http://hl7.org/fhir/MedicationStatement/example001>` vs. `[]` in the linked pages above.

This code checks if `base` has a value:

```java
  protected String getReferenceURI(String ref) {
    if (ref != null && (ref.startsWith("http://") || ref.startsWith("https://")))
      return "<" + ref + ">";
    else if (base != null && ref != null && ref.contains("/"))
      return "<" + Utilities.appendForwardSlash(base) + ref + ">";
    else
      return null;
    }
```

I think the calling code needs to provide a value for `base`. Not sure why it was in R4 but isn't now.

**Comment by balhoff** [comment]
I verified my idea above by hardcoding a base string. It fixes both the resource IRI and the `fhir:link` output. So we need to change the calling code. I see this is called in @tmprd's changes here, with a `null` value:

https://github.com/HL7/kindling/pull/148/files#diff-0b45ecac1a10eb5075600a7cb0da87b7531c1c2795f7d2519db5a50e1d440a42R4506

This must be called somewhere else as well, since R5 generation has been happening. I don't know enough about the full build.

**Comment by balhoff** [comment]
Discussed in WG meeting, if base is null, in the code default to base `http://hl7.org/fhir/` to match the R4 examples.

As a larger project, we think it would be nice if the example IRIs were dereferenced in a cleaner way, e.g. for something like `<http://hl7.org/fhir/MedicationStatement/example001>`, return a resource rather than redirect to `https://hl7.org/fhir/medicationstatementexample1.ttl`. (@ericprud)

**Comment by balhoff** [comment]
I opened a pull request for the above change: https://github.com/hapifhir/org.hl7.fhir.core/pull/1858

**Comment by dbooth-boston** [comment]
Done and merged: https://github.com/hapifhir/org.hl7.fhir.core/pull/1858
